drivers/dac/ltc2672/ltc2672.c

```diff
@@ -0,0 +1,532 @@
+	    || (current_reg_a > device->max_currents[out_ch])
+	    || (current_reg_b > device->max_currents[out_ch]))
+		return -EINVAL;
+
+	if (out_ch < LTC2672_DAC0 || out_ch > LTC2672_DAC4)
+		return -EINVAL;
+
+	/* Write code to Register A of out_ch */
+	current_code = (float)(current_reg_a / device->max_currents[out_ch]);
+
+	if (device->active_device == LTC2672_12)
+		current_code = (uint32_t)(current_code * (float)(LTC2672_12BIT_RESO)) << 4;
+	else
+		current_code *= LTC2672_16BIT_RESO;
+
+	command = LTC2672_COMMAND24_GENERATE(LTC2672_CODE_TO_CHANNEL_X, out_ch,(uint32_t)current_code);
+
+	ret = ltc2672_transaction(device, command, false);
+	if (ret)
+		return ret;
+
+	/* Toggle to enable register B*/
+	command = LTC2672_COMMAND24_GENERATE(LTC2672_TOGGLE_SEL, LTC2672_DAC0, NO_OS_BIT(out_ch));
+
+	ret = ltc2672_transaction(device, command, false);
+	if (ret)
+		return ret;
+
+	/* Write code to Register B of out_ch */
+	current_code = (float)(current_reg_b / device->max_currents[out_ch]);
+
+	if (device->active_device == LTC2672_12)
+		current_code = (uint32_t)(current_code * (float)(LTC2672_12BIT_RESO)) << 4;
+	else
+		current_code *= LTC2672_16BIT_RESO;
+
+	command = LTC2672_COMMAND24_GENERATE(LTC2672_CODE_TO_CHANNEL_X, out_ch,
+					     (uint32_t)current_code);
+
+	ret = ltc2672_transaction(device, command, false);
+	if (ret)
+		return ret;
+
+	/* Toggle to enable register A*/
+	command = LTC2672_COMMAND24_GENERATE(LTC2672_TOGGLE_SEL, LTC2672_DAC0, NO_OS_BIT(out_ch));
+
+	command &= ~NO_OS_BIT(out_ch);
+
+	return ltc2672_transaction(device, command, false);
+}
+
+/**
+ * @brief enables or disables selected channel/s for toggling
+ * @param device - ltc2672 descriptor
+ * @param mask - mask that describes which channel/s to toggle
+ * @return 0 in case of success, negative error code otherwise
+ */
+int ltc2672_enable_toggle_channel(struct ltc2672_dev *device, uint32_t mask)
+{
+	uint32_t command;
+
+	if (mask > 31)
+		return -EINVAL;
+
+	command = LTC2672_COMMAND24_GENERATE(LTC2672_TOGGLE_SEL, LTC2672_DAC0, mask);
+
+	return ltc2672_transaction(device, command, false);
+}
+
+/**
+ * @brief enables or disables global toggle bit
+ * @param device - ltc2672 descriptor
+ * @param is_enable - sets or resets the global toggle bit
+ * @return 0 in case of success, negative error code otherwise
+ */
+int ltc2672_global_toggle(struct ltc2672_dev *device, bool is_enable)
+{
+	uint32_t command;
+
+	device->global_toggle = is_enable;
+
+	command = LTC2672_COMMAND24_GENERATE(LTC2672_TOGGLE_GLBL, LTC2672_DAC0,
+					     (uint8_t)device->global_toggle);
+
+	return ltc2672_transaction(device, command, false);
+}
+
+/**
+ * @brief configures the fault detection bits
+ * @param device - ltc2672 descriptor
+ * @param mask - mask that describes desired configuration
+ * @return 0 in case of success, negative error code otherwise
+ */
+int ltc2672_config_command(struct ltc2672_dev *device, uint8_t mask)
+{
+	uint32_t command;
+
+	if (mask > 15)
```

Comment:
what's the meaning of `15`? Add a suggestive macro since all the defined masks in the header file seem to be greater than 15.